streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImplTest.java

```diff
@@ -96,4 +100,42 @@ public void testThroughputMetrics() {
         streamsMetrics.removeSensor(sensor1);
         assertEquals(defaultMetrics, streamsMetrics.metrics().size());
     }
+
+    @Test
+    public void testTotalMetricDoesntDecrease() {
+        final MockTime time = new MockTime();
+        final MetricConfig config = new MetricConfig().eventWindow(1).samples(2);
+        final Metrics metrics = new Metrics(config, time);
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, "");
+
+        final String scope = "scope";
+        final String entity = "entity";
+        final String operation = "op";
+
+        final Sensor sensor = streamsMetrics.addLatencyAndThroughputSensor(
+                scope,
+                entity,
+                operation,
+                Sensor.RecordingLevel.INFO
+        );
+
+        final double latency = 100.0;
+        final MetricName totalMetricName = metrics.metricName(
+                "op-total",
+                "stream-scope-metrics",
+                "",
+                "client-id",
+                "",
+                "scope-id",
+                "entity"
+        );
+
+        final KafkaMetric totalMetric = metrics.metric(totalMetricName);
+
+        for (int i = 0; i < 10; i++) {
+            assertEquals(i, totalMetric.measurable().measure(config, time.milliseconds()), 0.0001);
```

Comment:
why do we need a delta on this measurement? It seems like it would just be 0.0, 1.0, 2.0, etc?

Reply:
`assertEquals(double expected, double actual)` is deprecated

Reply:
super nit: the delta really doesn't matter in this case I'm wondering if `assertEquals(i, (int) totalMetric.measurable().measure(config, time.milliseconds()));` would be more clear, but this is subjective so feel free to ignore.